### PR TITLE
Add M5Stack Dial

### PR DESCRIPTION
**Pull request by jesserockz**
https://docs.m5stack.com/en/core/M5Dial

### CODE REVIEW

File: src/docs/devices/M5Stack-Dial/index.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+date-published: 2024-05-23
+type: misc
+standard: global
+board: esp32
+project-url: https://docs.m5stack.com/en/core/M5Dial
+---
+
+## Product Images
+
+![M5Stack Dial](M5Stack_Dial.png "M5Stack Dial")
```
Comment (jesserockz) [inline src/docs/devices/M5Stack-Dial/index.md]:
```suggestion
![M5Stack Dial](M5Stack-Dial.png "M5Stack Dial")
```

**Comment by nagyrobi** [comment]
This device would worth a LVGL example using `tileview`, that would resemble what can be seen on smartwatches.

**Comment by jesserockz** [comment]
> This device would worth a LVGL example using `tileview`, that would resemble what can be seen on smartwatches for navigation between the items...

We shouldn't add config for things that is not released in ESPHome, remember? =)

**Comment by nagyrobi** [comment]
That's exactly the reason why this display's config is kept in draft state here:
https://github.com/esphome/esphome-devices/pull/622

**Pull request merged**